pkg/monitoring/prometheusagent/sharding/sharding_test.go

```diff
@@ -0,0 +1,71 @@
+package sharding
+
+import (
+	"testing"
+)
+
+func TestShardComputationScaleUp(t *testing.T) {
+	pass := Strategy{ScaleUpSeriesCount: float64(1_000_000), ScaleDownPercentage: float64(0.20)}
```

Comment:
Nice test coverage, but for multiple test cases we usually wrap test cases into a slice and iterate over via t.Run